flag_engine/segments/models.py

```diff
@@ -20,6 +22,12 @@ class SegmentConditionModel:
 
     def matches_trait_value(self, trait_value: typing.Any) -> bool:
         # TODO: move this logic to the evaluator module
+        if (
+            type(trait_value) is str
+            and semver.VersionInfo.isvalid(trait_value)
+            and semver.VersionInfo.isvalid(self.value)
+        ):
+            trait_value = semver.VersionInfo.parse(trait_value)
```

Comment:
We're casting the trait_value to some kind of semver object here but not `self.value`. I think perhaps we need to update `get_casting_function` ? Although I'm not too sure how the tests are passing without it.

Reply:
I am glad you asked - It's one of the USP of the library(not really tho), i.e: you can pass a string to comparison functions, and it will parse that string object internally before doing the comparison

Reply:
Ah, ok, that's cool.